fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -766,6 +781,16 @@ private boolean checkTxnHasRelatedJob(TransactionState txnState, Map<Long, Set<L
             return true;
         }
 
+        if (txnState.getSourceType() == LoadJobSourceType.ROUTINE_LOAD_TASK) {
```

Comment:
It's not good to judge type here. You can put a callback when you begin a transaction, and then when transaction state is changed, your callback function will be called